### ISSUE TITLE
List Start Index 0 or 1?

### ISSUE THREAD

**Issue by liumiaowilson** [cesarParra/expression#130]
Hi, when I was using list related functions, something got me quite confused.

`SUBSTRING` function says that lists start index is 1-based, but `AT` function seems to be 0-based. Can you help clarify which is correct? Thank you for your help.

**Comment by cesarParra** [comment]
@liumiaowilson 

That's right, they work a little bit differently. At the beginning I was trying to match exactly how Salesforce formula fields and CRM Analytics formulas work, which a lot are 1 based, so they could be easily translatable. But then I introduced the list concept and kept it 0 based to match Apex.

I'll probably go back and clean up the very first few functions that got implemented to make sure things are consistent.
